packages/deepl-translation/server/deeplTranslate.js

```diff
@@ -0,0 +1,174 @@
+					{
+						'language': 'ES'
+					},
+					{
+						'language': 'IT'
+					},
+					{
+						'language': 'NL'
+					},
+					{
+						'language': 'PL'
+					}
+				];
+			}
+		}
+	}
+
+	translateMessage(message, room, targetLanguage) {
+		if (this.autoTranslateEnabled && this.apiKey) {
+			let targetLanguages;
+			if (targetLanguage) {
+				targetLanguages = [targetLanguage];
+			} else {
+				targetLanguages = RocketChat.models.Subscriptions.getAutoTranslateLanguagesByRoomAndNotUser(room._id, message.u && message.u._id);
+			}
+			if (message.msg) {
+				Meteor.defer(() => {
+					const translations = {};
+					let targetMessage = Object.assign({}, message);
+
+					targetMessage.html = s.escapeHTML(String(targetMessage.msg));
+					targetMessage = this.tokenize(targetMessage);
+
+					let msgs = targetMessage.msg.split('\n');
+					msgs = msgs.map(msg => encodeURIComponent(msg));
+					const query = `text=${ msgs.join('&text=') }`;
+
+					const supportedLanguages = this.getSupportedLanguages('en');
+					targetLanguages.forEach(language => {
+						if (language.indexOf('-') !== -1 && !_.findWhere(supportedLanguages, {language})) {
+							language = language.substr(0, 2);
+						}
+						let result;
+						try {
+							result = HTTP.get(this.apiEndPointUrl, {
+								params: {
+									auth_key: this.apiKey,
+									target_lang: language
+								}, query
+							});
+						} catch (e) {
+							console.log('Error translating message', e);
+							return message;
+						}
+						if (result.statusCode === 200 && result.data && result.data.translations && Array.isArray(result.data.translations) && result.data.translations.length > 0) {
+							// store translation only when the source and target language are different.
+							if (result.data.translations.map(translation => translation.detected_source_language).join() !== language) {
+								const txt = result.data.translations.map(translation => translation.text).join('\n');
+								translations[language] = this.deTokenize(Object.assign({}, targetMessage, {msg: txt}));
+							}
+						}
+					});
+					if (!_.isEmpty(translations)) {
+						RocketChat.models.Messages.addTranslations(message._id, translations);
+					}
+				});
+			}
+
+			if (message.attachments && message.attachments.length > 0) {
+				Meteor.defer(() => {
+					for (const index in message.attachments) {
+						if (message.attachments.hasOwnProperty(index)) {
+							const attachment = message.attachments[index];
+							const translations = {};
+							if (attachment.description || attachment.text) {
+								const query = `q=${ encodeURIComponent(attachment.description || attachment.text) }`;
+								const supportedLanguages = this.getSupportedLanguages('en');
+								targetLanguages.forEach(language => {
+									if (language.indexOf('-') !== -1 && !_.findWhere(supportedLanguages, {language})) {
+										language = language.substr(0, 2);
+									}
+									const result = HTTP.get(this.apiEndPointUrl, {
```

Comment:
Imho, only this sending the actual request should be implemented in the individual classes.
super-class: `translateMessage`
sub-class: `issueTranslateRequest`

Reply:
Handled